### ISSUE TITLE
Add option to allow unnecessary paretheses for Boolean expressions

### ISSUE THREAD

**Issue by alexbchr** [arturbosch/detekt#1592]
I think it would be great if the UnnecessaryParentheses rule could be a bit more customized, so we can allow unnecessary parentheses when building boolean expressions, especially when using AND and OR operators (`&&` and `||`).

If we take this example:
```kotlin
val easierToRead = (a && b) || (c && d)
val harderToRead = a && b || c && d
```

## Expected Behavior
The `easierToRead` expression should not trigger the `UnnecessaryParentheses` rule. This could be customized by an additional option on this rule.

## Current Behavior
The `easierToRead` expression is reported by the `UnnecessaryParentheses` rule, suggesting changing it to the `harderToRead` expression.

## Context
I think adding parentheses around OR (`||`) clauses in a boolean expression containing AND `&&`) clauses makes the expression easier to read and understand.

For reference, even the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s4.7-grouping-parentheses) recommends using grouping parentheses, as it improves readability.

**Comment by 3flex** [comment]
As mentioned by @schalkms [here](https://github.com/arturbosch/detekt/issues/1222#issuecomment-519887462) it's unlikely that this will be implemented. Strictly, these parentheses are unnecessary, so the rule is doing what it's designed to do. Neither official [Kotlin coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html) nor [Google's Kotlin style guide](https://developer.android.com/kotlin/style-guide) make mention of grouping parentheses.

In this case I'd suggest either disabling the rule for your codebase and adding/removing parentheses without static checks, or leaving the rule enabled and suppressing for specific functions where grouping parentheses make the intent of the code clearer.

cc @arturbosch in case his opinion differs :)

**Comment by schalkms** [comment]
@alexbchr if you really need this feature, you can implement a [custom rule](https://arturbosch.github.io/detekt/extensions.html#custom-rulesets) by copying the *default* rule and adapting it do your needs.
